### PR TITLE
Reintroduce light intensity scale factor removed in THREE r165 after WebGLRenderer.useLegacyLights deprecation (fix #5556, #5546)

### LINKED ISSUE #5556: Colors against dark in master vs 1.6.0. Probably related to THREE changes

### PR DESCRIPTION
**Pull request by dmarcos**
Is this good? @vincentfretin @mrxz

### CODE REVIEW

**Comment by dmarcos** [comment]
Colors of the hello world look same as before after applying the intensity scale factor

File: src/components/light.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ module.exports.Component = registerComponent('light', {
     color: {type: 'color', if: {type: ['ambient', 'directional', 'hemisphere', 'point', 'spot']}},
     envMap: {default: '', if: {type: ['probe']}},
     groundColor: {type: 'color', if: {type: ['hemisphere']}},
-    decay: {default: 1, if: {type: ['point', 'spot']}},
+    decay: {default: 2, if: {type: ['point', 'spot']}},
```
Comment (vincentfretin) [inline src/components/light.js:23]: Why did you change decay?

File: src/components/light.js [hunk 2]
```diff
@@ -281,7 +281,11 @@ module.exports.Component = registerComponent('light', {
     var distance = data.distance;
     var groundColor = new THREE.Color(data.groundColor);
     groundColor = groundColor.getHex();
-    var intensity = data.intensity;
+    // WebGLRenderer.useLegacyLights has been removed in THREE r165
+    // Lights intensity had an implicit scaleFactor of PI now removed.
+    // It's reintroduced here since it's easier to think about integers vs multiples of PI.
+    var intensityScaleFactor = Math.PI;
+    var intensity = data.intensity * intensityScaleFactor;
```
Comment (vincentfretin) [inline src/components/light.js:288]: It shouldn't change the intensity for scene that used `renderer="physicallyCorrectLights: true"` already.

**Review by vincentfretin — COMMENTED** [review]
We could keep the physicallyCorrectLights property on renderer just to add a condition in the light component. But we should remove the lines https://github.com/aframevr/aframe/blob/2aca98cbbc5f1f03431ce07f107709baa1718df8/src/systems/renderer.js#L38-L40 and modify the tests.

#5546 for context.

File: src/components/light.js (hunk 1 above)
Comment (dmarcos) [reply src/components/light.js:23]: This is @mrxz suggestion in https://github.com/aframevr/aframe/issues/5546#issuecomment-2191818981

File: src/components/light.js (hunk 2 above)
Comment (dmarcos) [reply src/components/light.js:288]: Not sure I understand the comment. Without the factor scene is dark. What should we do?

**Comment by dmarcos** [comment]
> We could keep the physicallyCorrectLights property on renderer just to add a condition in the light component. But we should remove the lines
> 
> https://github.com/aframevr/aframe/blob/2aca98cbbc5f1f03431ce07f107709baa1718df8/src/systems/renderer.js#L38-L40
> 
> and modify the tests.
> #5546 for context.

I think should be fine to remove the `physicallyCorrectLights` property in the renderer?

File: src/components/light.js (hunk 2 above)
Comment (vincentfretin) [reply src/components/light.js:288]: We have aframe experiences with renderer="physicallyCorrectLights:true" and light intensity that was carefully set, those scenes are not dark, we don't want the intensity to change for those.

**Comment by mrxz** [comment]
As @vincentfretin pointed out, there are already scenes that have `physicallyCorrectLights` enabled and should not have their intensities scaled. If we get rid of the property on `renderer`, then there won't be a way to distinguish the two.

Besides that, applying the scaling only in the `light` component will miss lights created in other ways. Components might directly create lights or a glTF file might include them. Even the `reflection` component manually sets light intensities, which would need to be updated as well ([reflection.js#L20](https://github.com/aframevr/aframe/blob/master/src/components/scene/reflection.js#L20)).

Instead of this PR, I think we should either:

1. Follow Three.js and deprecate the "legacy lights" entirely. The default light setup and things like the environment-component can be updated. Document what users need to do to upgrade their scenes.
2. Retain the old behaviour by re-introducing `useLegacyLights` in `super-three`. This ensures that everything behaves like before, instead of only the `light` component.

Personally not a fan of diverging from Three.js, but forcing upgrading users to make changes with no way to opt-out is also far from ideal.

**Comment by dmarcos** [comment]
Leaning towards deprecation

**Comment by dmarcos** [comment]
I multiplied the light defaults and values in examples by 3.14

https://github.com/aframevr/aframe/commit/8813d3d2818634ed4900409d99544b60587aa300

Also updated the environment component defaults and published 1.4.0

Any other outstanding 3rd party components that we should adjust?

**Comment by vincentfretin** [comment]
@dmarcos can you please also remove the physicallyCorrectLights option from the renderer system?

**Comment by dmarcos** [comment]
@vincentfretin done. thanks 🫡
